### ISSUE TITLE
TypeError: xxx.setExtras is not a function

### ISSUE THREAD

**Issue by cdvrooman** [kylestetz/CLNDR#305]
I am deleting a holiday from my database table with ajax and on success I delete that particular id from the days array and then update the list of "days" via calendar.setExtras();

Here is the (simplified) code used to instantiate clndr:

```
var calendar = $('#full-clndr').clndr({
	template: $('#full-clndr-template').html(),
	events: events,
	forceSixRows: true,
	constraints: {
		startDate: '2017-01-01'
	},
	extras: {
		days: {"70":"2017-04-28","71":"2017-04-29"}
	}
});

```

The easiest way to replicate the error is to immediately call:
calendar.setExtras();
or
calendar.setExtras({ days: {"71":"2017-04-29"} });
or
calendar.setExtras({});

All of them provoke the following error:

> TypeError: calendar.setExtras is not a function

instead of updating the extras array and rerendering the calendar.

I looked in the tests.js file, but there were no tests related to the setExtras() method.

I using XAMPP 5.6.30 on Windowx 10 64bit. Firefox Developer Edition, 54.0a2 (2017-04-16) (64-bit)

**Comment by criztovyl** [comment]
`setExtras` isn't yet released ([as of 1.4.7](https://github.com/kylestetz/CLNDR/compare/1.4.7...master#diff-04c6e90faac2675aa89e2176d2eec7d8R599)), make sure you use the files from the `master` branch or at least ebb0edb99bf9d1fe327d0a08d8c599ac78fde33b.

**Comment by mikegioia** [comment]
@cdvrooman Is this still an issue? It looks like `setExtras` is exposed publically.

cc @criztovyl (thanks for the first response)

**Comment by criztovyl** [comment]
As far as I can see, `setExtras` is still not released, see [this 1.4.7. vs master diff](https://github.com/kylestetz/CLNDR/compare/1.4.7...master#diff-04c6e90faac2675aa89e2176d2eec7d8R599).

I think the problem here could be that when you use bower or npm, by default you will only install the latest release, which is 1.4.7 atm, so you won't get `setExtras`.
Correct me if I'm wrong ^^

Also, `setExtras` *is* publically exposed in `clndr.js` *only*, but not in the `clndr.min.js` (wasn't updated since 1.4.7).

**Comment by criztovyl** [comment]
Re `clndr.min.js`: The problem could be that you refer to that file as the "production download", which may be misleading as I would expect the production version on the master branch as the current state-of-the-art, simply compressed.

**Comment by mikegioia** [comment]
Ok. I guess it must have missed the minification. I'll recompile it and push a new update.

**Comment by mikegioia** [comment]
This is resolved, 1.4.8 and 1.4.9 at least have it.
